element-connector/src/main/java/org/eclipse/californium/elements/Connector.java

```diff
@@ -91,4 +91,12 @@
 	 * @return the address
 	 */
 	public InetSocketAddress getAddress();
+
+	/**
+	 * Returns true if this is a TCP connector. TCP connector is subject to a different spec (CoAP-over-TCP) and
+	 * will requir CoAP endpoint to instantiate a different matcher, serializer and coap stack.
+	 *
+	 * @return true if this is a TCP based connector.
+     */
+	public boolean isTcp();
```

Comment:
Yes, I was actually rather unhappy with isTcp() method. Your proposed approach on connector sounds good to me (and makes DTLS work better).